t2t/data/dataset_readers/dataset_utils/contrastive_utils.py

```diff
@@ -58,17 +69,40 @@ def sample_anchor_positives(
     anchor = " ".join(tokens[anchor_start:anchor_end])
 
     # Sample positives from around the anchor. The intuition being that text that appears close
-    # together is the same document is likely to be semantically similar. These may be adjacent or
-    # overlap with each other and the anchor. Their length is sampled from a beta distribution
-    # skewed towards shorter spans.
+    # together is the same document is likely to be semantically similar. By default, these may be
+    # adjacent or overlap with each other and the anchor, but this can be selected manually with
+    # the `strategy` parameter. Their length is sampled from a beta distribution skewed towards
+    # shorter spans.
     positives = []
     for _ in range(num_spans):
         positive_length = int(np.random.beta(2, 4) * (max_span_len - min_span_len) + min_span_len)
-        # Be careful not to run off the edges of the document, as this error will pass silently.
-        positive_start = randint(
-            max(0, anchor_start - positive_length), min(anchor_end, num_tokens - positive_length),
-        )
-        positive_end = positive_start + positive_length
-        positives.append(" ".join(tokens[positive_start:positive_end]))
+        if strategy and strategy == strategy.subsuming:
+            positive_start = randint(anchor_start, anchor_end - positive_length)
+        elif strategy and strategy == strategy.adjacent:
+            # There are two types of adjacent positives, those that boarder the beginning of the
+            # anchor and those that boarder the end. These may not be valid in cases where the
+            # positive length is longer than the available space in the document.
+            valid_starts = []
+            if anchor_start - positive_length > 0:
+                valid_starts.append(anchor_start - positive_length)
+            elif anchor_end + positive_length <= num_tokens:
+                valid_starts.append(anchor_end + positive_length)
+            else:
+                raise ValueError(
+                    (
+                        f"Cannot select adjacent only spans for a span length of {positive_length}"
+                        f" and a number of tokens len({tok_method}) ({num_tokens})."
+                    )
+                )
+            positive_start = choice(valid_starts)
+        else:
+            # Be careful not to run off the edges of the document, as this error will pass silently.
+            positive_start = randint(
+                max(0, anchor_start - positive_length),
+                min(anchor_end, num_tokens - positive_length),
+            )
+
+    positive_end = positive_start + positive_length
+    positives.append(" ".join(tokens[positive_start:positive_end]))
```

Comment:
@OsvaldN Do you mind sanity checking this? It will allow us to perform the ablations as discussed.

Reply:
I changed adjacent sampling, otherwise looks great

Reply:
Nice, thanks! Exactly why I asked haha